crux_core/src/bridge/mod.rs

```diff
@@ -44,35 +42,66 @@ where
     /// Create a new Bridge using the provided `core`.
     pub fn new(core: Core<Eff, A>) -> Self {
         Self {
-            inner: BridgeWithSerializer::new(core, Bincode),
+            inner: BridgeWithSerializer::new(core),
         }
     }
 
     /// Receive an event from the shell.
     ///
     /// The `event` is serialized and will be deserialized by the core before it's passed
     /// to your app.
-    pub fn process_event<'de>(&self, event: &'de [u8]) -> Vec<u8>
+    pub fn process_event(&self, event: &[u8]) -> Vec<u8>
     where
-        A::Event: Deserialize<'de>,
+        A::Event: for<'a> Deserialize<'a>,
     {
-        self.inner.process_event(event)
+        let mut return_buffer = vec![];
+        let mut deser = bincode::Deserializer::from_slice(
+            event,
+            DefaultOptions::new()
+                .with_fixint_encoding()
+                .allow_trailing_bytes(),
+        );
+        self.inner.process_event(
+            &mut deser,
+            &mut bincode::Serializer::new(&mut return_buffer, DefaultOptions::new()),
```

Comment:
This probably shouldn't be default options actually, but the `with_fixint_encoding` I copied from above started complaining and I couldn't be bothered figuring it out

Reply:
Should they both be default options instead...? Do we have opinions on the bincode serialization?

Reply:
These are the options that `bincode::deserialize` sets up, so I'm just keeping things consistent.  Bit weird of that they don't use default options by default, but... y'know

Reply:
Although looks like I didn't push up the change where I fixed this. hmm.

Reply:
Ok, options are now consistently the same as those used by `bincode::deserialize`